src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2022 Kuali, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.kuali.kfs.kew.api.document;
+
+public interface DocumentProcessingQueue {
+
+    void process(String documentId) throws IllegalArgumentException;
+
+    void processWithOptions(String documentId, DocumentProcessingOptions options) throws IllegalArgumentException;
+
+    /*
+     * CU Customization: Added a separate method for handling the FINP-8541 functionality (which checks whether
+     * a document has been routed and, if not, will add a Complete request if one doesn't exist). This is needed
+     * to fix a DB transaction issue, where merely retrieving the route header can cause it to remain stale
+     * when subsequently invoking the workflow engine.
+     */
+    default boolean shouldProceedWithDocumentProcessing(String documentId) {
+        throw new IllegalStateException("Method has not been overridden yet; this default implementation "
```

Comment:
I don't understand why we want to throw this exception.  If the method hasn't been overriden, won't that break routing

Reply:
I was trying to find a way to avoid breaking other potential compilation problems while still indicating that it would be best to explicitly implement this method. However, I agree that the thrown exception is confusing, so I'll update this method to default to true.